### ISSUE TITLE
[Cache] TLS support on RedisTrait

### ISSUE THREAD

**Issue by c-lambert** [symfony/symfony#40224]
Hello,

rediss:// DSL scheme is for TLS connexion to Redis Server. 
You support rediss keyword but not TLS connexion.

https://github.com/predis/predis
```
$client = new Predis\Client('tls://127.0.0.1');
```

https://github.com/phpredis/phpredis
```
$redis->connect('tls://127.0.0.1', 6379); // enable transport level security.
```

When i use rediss:// scheme, the Redis connection still use tcp scheme.

**Comment by stof** [comment]
which RedisTrait are you talking about ? We have support for Redis in multiple components

**Comment by stof** [comment]
OK, this is in symfony/cache, because the initial merged PR was incomplete (which I reported back then, but I looked at it only at the time of being merged): https://github.com/symfony/symfony/pull/30605

**Comment by nicolas-grekas** [comment]
Might be fixed by https://github.com/symfony/symfony/pull/39599?

**Comment by c-lambert** [comment]
> Might be fixed by #39599?

Thank you Nicolas. It seems to be treated by this PR but still not merged.

I try to create a custom "app.redis_provider" but don't find any documention. I want to use my extended RedisAdapter, which init my own \Redis|\RedisArray|\RedisCluster|\Predis\ClientInterface and use it on cache.yaml configuration

**Comment by c-lambert** [comment]
I did this

```php
<?php


namespace App\Cache;


use Predis\Client;
use Symfony\Component\Cache\Adapter\RedisAdapter;
use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;

class RedisCacheAdapter extends RedisAdapter
{
    public function __construct(
        ParameterBagInterface $parameterBag
    ) {
        $redisScheme = $parameterBag->get('redis_scheme');
        $redisHost = $parameterBag->get('redis_host');
        $redisPort = $parameterBag->get('redis_port');
        $redisKey = $parameterBag->get('redis_key');
        $client = new Client(
            [
                'scheme' => $redisScheme,
                'host' => $redisHost,
                'port' => $redisPort,
                'password' => $redisKey
            ]
        );
        parent::__construct($client);
    }
}
```

and override cache.app in service.yaml
```yaml
services:
    cache.app:
        class: App\Cache\RedisCacheAdapter
        arguments:
            - '' # i don't know why
```
